builtin/credential/ldap/path_login.go

```diff
@@ -103,6 +103,10 @@ func (b *backend) pathLogin(ctx context.Context, req *logical.Request, d *framew
 		},
 	}
 
+	if cfg.UsernameAsAlias && cfg.UserFilter == "" {
```

Comment:
If we don't allow `cfg` to get into this state on `NewConfigEntry` we could just check against `cfg.UsernameAsAlias`.

Reply:
The UX of this is annoying because `userfilter` has a default. Considering dropping it 🤔